crates/sui-types/src/error.rs

```diff
@@ -514,9 +511,24 @@ impl From<SubscriberError> for SuiError {
     }
 }
 
-impl From<tonic::Status> for SuiError {
-    fn from(status: tonic::Status) -> Self {
-        Self::RpcError(status.message().to_owned(), status.code().description())
+impl From<Status> for SuiError {
+    fn from(status: Status) -> Self {
+        let result = bincode::deserialize::<SuiError>(status.details());
+        if let Ok(sui_error) = result {
+            sui_error
+        } else {
+            Self::RpcError(
+                status.message().to_owned(),
+                status.code().description().to_owned(),
```

Comment:
do we need `to_owned` here?

Reply:
Yes description() returns `static str, but here we need String for proper deserialization support